web-bundle/src/main/java/com/graphhopper/replica/ReplicaCustomSpeedsFootTagParser.java

```diff
@@ -31,24 +37,31 @@ public void handleWayTags(int edgeId, EdgeIntAccess edgeIntAccess, ReaderWay way
         if (highwayValue == null) {
             if (way.hasTag("route", ferries)) {
                 double ferrySpeed = ferrySpeedCalc.getSpeed(way);
-                setSpeed(edgeId, edgeIntAccess, true, true, ferrySpeed, way);
+                setSpeed(edgeId, edgeIntAccess, ferrySpeed, way);
             }
             if (!way.hasTag("railway", "platform") && !way.hasTag("man_made", "pier"))
                 return;
         }
 
         String sacScale = way.getTag("sac_scale");
         if (sacScale != null) {
-            setSpeed(edgeId, edgeIntAccess, true, true, "hiking".equals(sacScale) ? MEAN_SPEED : SLOW_SPEED, way);
+            setSpeed(edgeId, edgeIntAccess, "hiking".equals(sacScale) ? MEAN_SPEED : SLOW_SPEED, way);
         } else {
-            setSpeed(edgeId, edgeIntAccess, true, true, way.hasTag("highway", "steps") ? MEAN_SPEED - 2 : MEAN_SPEED, way);
+            setSpeed(edgeId, edgeIntAccess,  way.hasTag("highway", "steps") ? MEAN_SPEED - 2 : MEAN_SPEED, way);
         }
     }
 
     // Overrides speed with custom speed, if one exists
-    void setSpeed(int edgeId, EdgeIntAccess edgeIntAccess, boolean fwd, boolean bwd, double speed, ReaderWay way) {
-         double speedToSet = CustomSpeedsUtils.getCustomMaxSpeed(way, osmWayIdToDirectedMaxSpeeds, bwd).orElse(speed);
-         setSpeed(edgeId, edgeIntAccess, fwd, bwd, speedToSet);
+    void setSpeed(int edgeId, EdgeIntAccess edgeIntAccess, double speed, ReaderWay way) {
+        if (getCustomSpeedIdType(idToCustomSpeeds) == CustomSpeedId.IdType.SEID) {
+            throw new NotImplementedException("SEID-based custom speeds are not yet implemented for foot!");
+        } else {
+            ImmutableMap<Long, DirectedSpeeds> osmWayIdToDirectedMaxSpeeds = getOsmBasedCustomSpeeds(idToCustomSpeeds);
+            double forwardSpeed = CustomSpeedsUtils.getCustomMaxSpeed(way, osmWayIdToDirectedMaxSpeeds, false).orElse(speed);
```

Comment:
Note: we're refactoring a bit here to make the things cleaner (this function was always called with `fwd` and `bwd` set to true, so we could remove those params) but lines 60 + 61 are also fixing what was technically incorrect logic. 

Before, we were only grabbing the osm-based custom speed in the forward direction, and applying it to both directions of the edge if it was present; now, we grab the forward and backward custom speds, and apply them appropriately in each separate direction